backend/pkg/api/data_access/vdb_blocks.go

```diff
@@ -32,319 +41,272 @@ func (d *DataAccessService) GetValidatorDashboardBlocks(ctx context.Context, das
-	var val any
-	sortColName := `slot`
+	var offset any
 	switch colSort.Column {
-	case enums.VDBBlockProposer:
-		sortColName = `proposer`
-		val = currentCursor.Proposer
-	case enums.VDBBlockStatus:
-		sortColName = `status`
-		val = currentCursor.Status
-	case enums.VDBBlockProposerReward:
-		sortColName = `el_reward + cl_reward`
-		val = currentCursor.Reward
-	}
-	onlyPrimarySort := sortColName == `slot`
-	if currentCursor.IsValid() {
-		sign := ` > `
-		if colSort.Desc && !currentCursor.IsReverse() || !colSort.Desc && currentCursor.IsReverse() {
-			sign = ` < `
-		}
-		if currentCursor.IsReverse() {
-			if sortOrder == ` ASC` {
-				sortOrder = ` DESC`
-			} else {
-				sortOrder = ` ASC`
-			}
-		}
-		params = append(params, currentCursor.Slot)
-		where += `WHERE (`
-		if onlyPrimarySort {
-			where += `slot` + sign + fmt.Sprintf(`$%d`, len(params))
-		} else {
-			params = append(params, val)
-			secSign := ` < `
-			if currentCursor.IsReverse() {
-				secSign = ` > `
-			}
-			if sortColName == "status" {
-				// explicit cast to int because type of 'status' column is text for some reason
-				sortColName += "::int"
-			}
-			where += fmt.Sprintf(`(slot`+secSign+`$%d AND `+sortColName+` = $%d) OR `+sortColName+sign+`$%d`, len(params)-1, len(params), len(params))
+	case enums.VDBBlocksColumns.Proposer:
+		offset = currentCursor.Proposer
+	case enums.VDBBlocksColumns.Block:
+		offset = currentCursor.Block
+		if !currentCursor.Block.Valid {
+			offset = nil
 		}
-		where += `) `
-	}
-	if sortOrder == ` ASC` {
-		sortOrder += ` NULLS FIRST`
-	} else {
-		sortOrder += ` NULLS LAST`
+	case enums.VDBBlocksColumns.Status:
+		offset = fmt.Sprintf("%d", currentCursor.Status) // type of 'status' column is text for some reason
+	case enums.VDBBlocksColumns.ProposerReward:
+		offset = currentCursor.Reward
 	}
-	orderBy += sortColName + sortOrder
-	secSort := `DESC`
-	if !onlyPrimarySort {
-		if currentCursor.IsReverse() {
-			secSort = `ASC`
-		}
-		orderBy += `, slot ` + secSort
+
+	order, directions := applySortAndPagination(defaultColumns, t.SortColumn{Column: colSort.Column.ToExpr(), Desc: colSort.Desc, Offset: offset}, currentCursor.GenericCursor)
+	blocksDs = goqu.From(blocksDs). // encapsulate so we can use selected fields
+					Order(order...)
+	if directions != nil {
+		blocksDs = blocksDs.Where(directions)
 	}
 
-	// Get scheduled blocks. They aren't written to blocks table, get from duties
-	// Will just pass scheduled proposals to query and let db do the sorting etc
-	var scheduledProposers []t.VDBValidator
-	var scheduledEpochs []uint64
-	var scheduledSlots []uint64
-	// don't need to query if requested slots are in the past
+	// 5. Gather and supply scheduled blocks to let db do the sorting etc
 	latestSlot := cache.LatestSlot.Get()
-	if !onlyPrimarySort || !currentCursor.IsValid() ||
+	onlyPrimarySort := colSort.Column == enums.VDBBlockSlot
+	if !(onlyPrimarySort || colSort.Column == enums.VDBBlockBlock) || !currentCursor.IsValid() ||
```

Comment:
**question:** As discussed I don't understand how this if works.

Reply:
Couldn't replicate